### PR TITLE
174 feature oai

### PR DESCRIPTION
**Pull request by MartinPaulEve**
This completes the OAI implementation as per OCLC's request

### CODE REVIEW

File: src/templates/common/apis/OAI.xml [hunk 1]
```diff
@@ -4,8 +4,8 @@
     xsi:schemaLocation="http://www.openarchives.org/OAI/2.0/
         http://www.openarchives.org/OAI/2.0/OAI-PMH.xsd">
  <responseDate>{% now "Y-m-d\TH:i:s\Z" %}</responseDate>
-    <request verb="ListRecords" id="{{ journal_settings.general.publisher_name }}" metadataPrefix="oai_dc">{% if request.is_secure %}https://{% else %}http://{% endif %}{{ journal_settings.general.base_url }}/oai</request>
-    <ListRecords>       
+    <request verb="ListRecords" id="{{ journal_settings.general.publisher_name }}" metadataPrefix="oai_dc">http://{{ request.get_host }}{% url 'OAI_list_records' %}</request>
```
Comment (ajrbyers) [inline src/templates/common/apis/OAI.xml:7]: Could probably use the `{% journal_url %}` tag here.

Comment (ajrbyers) [reply src/templates/common/apis/OAI.xml:7]: Also we shouldn't force http.

File: src/templates/common/apis/OAI.xml [hunk 2]
```diff
@@ -21,18 +21,21 @@
                     xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
                     xsi:schemaLocation="http://www.openarchives.org/OAI/2.0/oai_dc/
                     http://www.openarchives.org/OAI/2.0/oai_dc.xsd">
-                    <dc:title xml:lang="en-US">{{article.title}}</dc:title>
+                    <dc:title>{{article.title}}</dc:title>
```
Comment (ajrbyers) [inline src/templates/common/apis/OAI.xml:24]: Should we strip tags from title?

**Comment by mauromsl** [comment]
test this please

File: src/templates/common/apis/OAI.xml [hunk 3]
```diff
@@ -21,18 +21,21 @@
...
                     {% for author in article.authors.all %}
                     <dc:creator>{{author.last_name}}, {{author.first_name}}</dc:creator>
                     {% endfor %}
-                    <dc:subject xml:lang="en-US"></dc:subject>
-                    <dc:description xml:lang="en-US">{{article.abstract}}</dc:description>
-                    <dc:publisher xml:lang="en-US">{{ request.journal.publisher }}</dc:publisher>
+                    <dc:description>{{article.abstract}}</dc:description>
```
Comment (ajrbyers) [inline src/templates/common/apis/OAI.xml:28]: Should we strip tags from abstract?

File: src/templates/common/apis/OAI.xml [hunk 4]
```diff
@@ -21,18 +21,21 @@
...
+                    <dc:publisher>{{ request.journal.publisher }}</dc:publisher>
                     <dc:date>{{article.date_published}}</dc:date>
-                    <dc:type xml:lang="en-US">Article</dc:type>
-                    <dc:type xml:lang="en-US"></dc:type>
-                    <dc:identifier>{{article.identifier}}</dc:identifier>
-                    <dc:coverage xml:lang="en-US"></dc:coverage>
+                    <dc:type>info:eu-repo/semantics/article</dc:type>
+                    <dc:volume>{{ article.issue.volume }}</dc:volume>
```
Comment (ajrbyers) [inline src/templates/common/apis/OAI.xml:32]: Its worth noting it *is* possible `article.issue` will be `None`

File: src/templates/common/apis/OAI.xml [hunk 5]
```diff
@@ -21,18 +21,21 @@
...
+                    <dc:issue>{{ article.issue.issue }}</dc:issue>
+                    <dc:identifier>{% if article.identifier.id_type == 'doi' %}info:doi/{% endif %}{{article.identifier.identifier}}</dc:identifier>
+                    <dc:source>{{ request.journal.name}}</dc:source>
+                    <dc:source>{{ request.journal.issn}}</dc:source>
```
Comment (ajrbyers) [inline src/templates/common/apis/OAI.xml:36]: ISSN can also be `None`

**Comment by MartinPaulEve** [comment]
Waiting for a response from OCLC to confirm which tags we're supposed to use

**Pull request merged**